Assets/Mirage/Runtime/InterestManagement/InterestManager.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Mirage.InterestManagement
+{
+    /// <summary>
+    /// An interest manager is responsible for showing and hiding objects to players
+    /// based on an area if interest policy.
+    ///
+    /// Implement this class to provide a interest management policy and assign
+    /// it to the <see cref="ServerObjectManager"/>
+    /// </summary>
+    public abstract class InterestManager : MonoBehaviour
```

Comment:
I think it would be better to have something like this, so that the monobehaviour part is separated from the InterestManager part
```cs
public abstract class InterestManagerCreator : MonoBehaviour 
{
    public abstract InterestManager CreateInterestManager();
}

public abstract class InterestManager
{
    // .. AOI methods here
}
```

Reply:
But what exactly is the point?  now to create an interest manager you will have to extend 2 classes where one of them just creates the other.

Reply:
- It doesn't need to be a `MonoBehaviour`, therefore it shouldn't be one
- Later when we move to tick based system `NetworkServer` (or something else) will be calling Tick/Update on `InterestManager`. Having it as a `MonoBehaviour` may lead people to use Unity's callback when they shouldn't be used.
- it not being a `Monobehaviour` makes it easier to support .net core (minor issue)